File: server/populate.js
const axios = require('axios');
const sleep = require('sleep');

axios.defaults.withCredentials = true;
const url = 'http://127.0.0.1:8000/services/client';


const loginUrl = 'http://127.0.0.1:8000/login';
const data = {
    username: 'admin',
    password: 'admin123',
};
axios(loginUrl, { method: 'POST', data }).then(res => {
    console.log(res.data);
    sleep.msleep(2000);
    for (let j = 0; j < 50; j++) {
        for (let i = 0; i <= 200; i++) {
            const data2 = {
                identifier: `popa${j}${i}`,
                lang: 'en',
                name: `popa${j}${i}`,
                type: 'demo',

            };
            axios({
                method: 'post',
                url,
                data: data2,
            }).then(res => {
                console.log(res);
                sleep.msleep(2000);
                if (i % 50 === 0) {
                    sleep.msleep(2000);
                }
            }, err => {
                console.log(err.message);
            });
        }
    }
}, err => {
    console.log(err);
});

File: tools/populate.py
import json
import time
from requests import Request, Session
url = 'http://127.0.0.1:8000/api/client';
loginUrl = 'http://127.0.0.1:8000/login';
s = Session()
r = Request('POST',loginUrl, data = {'username':'admin','password':'admin123'})

preppend = r.prepare()


resp = s.send(preppend)

for i in range(1):
    for j in range(1):
        ele = {
            'identifier': 'popa_'+str(i)+'_'+str(j),
            'lang': 'en',
            'name': 'popa_'+str(i)+'_'+str(j),
            'type': 'demo',

        }
        headers = {'Content-type': 'application/json', 'Accept': 'text/plain'}
      
        
        resp = s.post(url, data =json.dumps(ele), headers=headers)
        print(resp)


